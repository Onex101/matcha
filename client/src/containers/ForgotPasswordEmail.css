@media all and (min-width: 480px) {
    .ForgotEmail {
      padding: 60px 0;
    }
  
    .ForgotEmail form {
      margin: 0 auto;
      max-width: 500px;
    }
  }
  #emailSent {
      margin: 5vh auto;
      text-align: center;
  }