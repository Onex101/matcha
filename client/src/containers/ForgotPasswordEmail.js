import React, { Component } from "react";
import { Button, FormGroup, FormControl, ControlLabel } from "react-bootstrap";
import "./ForgotPasswordEmail.css";

export default class ForgotPasswordEmail extends Component {
    constructor(props) {
        super(props);

        this.state = {
            email: "",
            emailSent: false
        };
    }

    validateForm() {
        return this.state.email.length > 0;
    }

    handleChange = event => {
        this.setState({
            [event.target.id]: event.target.value
        });
    }

    validateEmail(mail) {
        var re = /\S+@\S+\.\S+/;
        if (re.test(mail)) {
            // console.log(mail)
            return (true)
        }
        alert("You have entered an invalid email address!")
        return (false)
    }

    handleSubmit = async event => {
        event.preventDefault();
        if (this.validateEmail(this.state.email)) {
            //   Verify and Send password reset email

            //Tutorial: https://itnext.io/password-reset-emails-in-your-react-app-made-easy-with-nodemailer-bb27968310d7
            try {
                const data = this.state;
                fetch(`/email/check`, {
                    method: "POST",
                    headers: {
                        "Content-Type": "application/json; charset=utf-8",
                    },
                    body: JSON.stringify({
                        email: data.email
                    })
                })
                    .then(response => response.json())
                    .then((responseJSON) => {
                        console.log(responseJSON);
                        if (responseJSON["error"] === null) {
                            this.setState({ emailSent: true });
                        }
                        else
                            alert("Something went wrong :(\n" + responseJSON["error"]);
                    })
                    .catch(err => console.error(err))
            } catch (e) {
                alert(e.message);
            }
            this.setState({ emailSent: true });
        }
    }

    render() {
        return (
            <div>
                {!this.state.emailSent
                    ? <div data-testid="notSent" className="ForgotEmail">
                        <form onSubmit={this.handleSubmit}>
                            <FormGroup controlId="email" bsSize="large">
                                <ControlLabel>Email</ControlLabel>
                                <FormControl
                                    data-testid="emailInput"
                                    autoFocus
                                    type="text"
                                    value={this.state.email}
                                    onChange={this.handleChange}
                                >
                                </FormControl>
                                
                                </FormGroup>
                            <Button
                                block
                                bsSize="large"
                                disabled={!this.validateForm()}
                                type="submit">
                                Send password reset email</Button>
                        </form>
                    </div>
                    : <div data-testid="sent" id="emailSent">
                        <p>An email has been sent to set up a new password.</p>
                    </div>}
            </div>

        );
    }
}
