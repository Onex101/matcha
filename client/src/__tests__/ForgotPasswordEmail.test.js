import React from 'react';
import {render, cleanup, fireEvent,} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ForgotPasswordEmail from '../containers/ForgotPasswordEmail.js';

afterEach(cleanup);

describe("Testing Forgot Password Email", ()=>{
	it("Display Notifications", ()=>{
		const { asFragment, getByText, getByTestId, getByLabelText } = render(<ForgotPasswordEmail />);
		expect(getByTestId("notSent")).toHaveTextContent("Send password reset email");

		const contentInput = getByTestId("emailInput");
		expect(contentInput.value).toBe('')
		fireEvent.change(contentInput, {
			target: { value: 'test@gmail.com' }
		})
		expect(contentInput.value).toBe('test@gmail.com')
		fireEvent.click(getByText("Send password reset email"))
		expect(getByTestId("sent")).toHaveTextContent("An email has been sent to set up a new password.");

		expect(asFragment()).toMatchSnapshot();
	});
})
