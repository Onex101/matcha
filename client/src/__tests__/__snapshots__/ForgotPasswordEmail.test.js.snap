// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Forgot Password Email Display Notifications 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="sent"
      id="emailSent"
    >
      <p>
        An email has been sent to set up a new password.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
